spec/controllers/reviews_controller_spec.rb

```diff
@@ -3,58 +3,51 @@
 describe ReviewsController do
   describe "POST create" do
     let(:video) { Fabricate(:video) }
+    let(:current_user) { Fabricate(:user) }
+    before { set_current_user(current_user) }
 
-    context "with authenticated users" do
-
-      let(:current_user) { Fabricate(:user) }
-      before { session[:user_id] = current_user.id }
+    it_behaves_like "requires sign in" do
+      let(:action) { post :create, review: Fabricate.attributes_for(:review), video_id: video.id }
+    end
 
-      context "with valid inputs" do
-        before do
-          post :create, review: Fabricate.attributes_for(:review), video_id: video.id
-        end
+    context "with valid inputs" do
+      before do
+        post :create, review: Fabricate.attributes_for(:review), video_id: video.id
+      end
 
-        it "redirects to the video show page" do
-          expect(response).to redirect_to video
-        end
-        it "creates a review" do
-          expect(Review.count).to eq(1)
-        end
-        it "creates a review associated with video" do
-          expect(Review.first.video).to eq(video)
-        end
-        it "creates a review associated with signed in user" do
-          expect(Review.first.user).to eq(current_user)
-        end
+      it "redirects to the video show page" do
+        expect(response).to redirect_to video
+      end
+      it "creates a review" do
+        expect(Review.count).to eq(1)
+      end
+      it "creates a review associated with video" do
+        expect(Review.first.video).to eq(video)
+      end
+      it "creates a review associated with signed in user" do
+        expect(Review.first.user).to eq(current_user)
       end
+    end
 
-      context "with invalid inputs" do
-        it "does not create a review" do
-          post :create, review: {rating: 4}, video_id: video.id
-          expect(Review.count).to eq(0)
-        end
+    context "with invalid inputs" do
+      it "does not create a review" do
+        post :create, review: {rating: 4}, video_id: video.id
```

Comment:
What else is required? For the purpose of creating a rating from the queue index, shouldn't a rating and video id be enough?